dev_tools/utils/github.py

```diff
@@ -0,0 +1,89 @@
+import requests
+
+def format_author_name(nm):
+    """
+    make sure all words in name start with a capital
+    """
+    s = ""
+    nms = nm.split(" ")
+    for n in nms:
+        s += n[0].upper() + n[1:] + " "
+    s = s[:-1]
+    return(s)
+
+def get_github_profile_name(email):
+    """
+    given an email addr return the github login name
+    """
+    email = email.replace(' ', '')
+    idx1 = email.find("@")
+    return(email[:idx1])
+
+def get_file_contributors_via_api(repo_name, file_path):
+    """
+    This function takes the repo name (ex:"definitions") and relative path to the nxdl
+    file (ex: "applications/NXmx.nxdl.xml") and using the github api it retrieves a dictionary
+    of committers for that file in descending date order.
+
+    In order to increase the capacity (rate) of use of the github API an access token is used if it exists
+    as an environment variable called GH_TOKEN, in the ci yaml file this is expected to be assigned from the secret
+    object like this
+    env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+
+    With the access token the rate is 5000 times per hour and without it is 60
+
+    returns a sorted dict of unique contributors to a file, or None if no GH_TOKEN has been defined in os.environ
+    """
+    have_token = False
+    if "GH_TOKEN" in os.environ.keys():
+        access_token = os.environ["GH_TOKEN"]
+        if len(access_token) > 0:
+            have_token = True
+    else:
+        # because the environment does not contain GH_TOKEN, assume the user wants to build the
+        # docs without contributor info
+        return(None)
+
+    contrib_skip_list = ["GitHub"]
+    url = f"https://api.github.com/repos/nexusformat/{repo_name}/commits"
+    params = {"path": file_path}
+    headers = {}
+    if have_token:
+        # Set the headers with the access token
+        headers = {
+            'Authorization': f'token {access_token}',
+            'Accept': 'application/vnd.github.v3+json'
+        }
+
+    response = requests.get(url, params=params, headers=headers)
+    commits = response.json()
+    if response.status_code == 403:
+        # the max rate per hour has been reached
+        raise Exception(f"{commits['message']},{commits['documentation_url']}")
+
+    contributor_names = set()
+    contribs_dct = {}
+    _email_lst = []
+    for commit_dct in commits:
+        if commit_dct["committer"] is not None:
+            contributor = commit_dct["commit"]["committer"]["name"]
+            if contributor in contrib_skip_list:
+                continue
+            contributor_names.add(contributor)
+            if commit_dct["commit"]["committer"]["email"] not in _email_lst:
+                _email = commit_dct["commit"]["committer"]["email"]
+                _email_lst.append(_email)
+                contribs_dct[commit_dct["commit"]['committer']["date"]] = {
+                    "name": format_author_name(commit_dct["commit"]["committer"]["name"]), "committer": commit_dct}
```

Comment:
Pass back smaller dictionary with just items you need.

Reply:
my thought here was that down the road if we wanted to add for example an **Authors** section or something else all the information would already be there and would just need to be referenced during the rst line generation in nxdl.py,

Reply:
Okay, that's fine. Perhaps change the key used to `"commit_dct"`.

Reply:
done!